pallets/identity/src/lib.rs

```diff
@@ -1702,25 +1706,33 @@ impl<T: Trait> Module<T> {
     /// # Errors
     /// - 'ConfidentialScopeClaimNotAllowed` if :
     ///     - Sender is not the issuer. That claim can be only added by your-self.
-    ///     - You are not the owner of that CDD_ID.
+    ///     - If claim is not valid.
     ///
     fn base_add_confidential_scope_claim(
         target: IdentityId,
         claim: Claim,
         issuer: IdentityId,
         expiry: Option<T::Moment>,
-        cdd_id: CddId,
     ) -> DispatchResult {
-        // Only onwer of the identity can add that confidential claim.
+        // Only owner of the identity can add that confidential claim.
         ensure!(
             issuer == target,
             Error::<T>::ConfidentialScopeClaimNotAllowed
         );
 
-        // Verify the onwer of that CDD_ID.
+        // Ensure scope should be `Ticker` for this claim.
         ensure!(
-            Self::base_fetch_cdd(target, T::Moment::zero(), Some(cdd_id)).is_some(),
-            Error::<T>::ConfidentialScopeClaimNotAllowed
+            claim
+                .as_scope()
+                .cloned()
```

Comment:
`scope_type` takes `&self`, so this doesn't seem necessary? Also, `.filter(...).is_some()` would be more appropriate here semantically imo.